### PR TITLE
[CI] error codes utility runner workflow

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
Signed-off-by: Rudraksh Pareek rudrakshpareek3601@gmail.com

**Description**

This PR fixes #

**Notes for Reviewers**
Updates the error ref publisher. Now it'll also update Go files. No one will ever need to care about what integer they should assign to an error code or updating it to fix merge conflicts if this works out well.
What changes this PR brings forth:
* Remove the `errorutil_*` json files generated by the utility to helpers dir and also `.gitignore` because we actually don't need to track them.
* A workflow which runs whenever changes in go code happen.
* **Expects error codes are assigned alphabetic strings (placeholders) when any code is merged to master.**
* The workflow will update the placeholders to integer codes based on `next_error_code` and also update `component_info.json` afterwards.

### UPDATED (with changes stolen from meshery#3528 )
**To explain what will happen, the entire procedure with diffs while testing this:**
1. Go code gets merged into the mater branch: https://github.com/DelusionalOptimist/meshery/commit/5c6c470c8a32d0a168da9f13c71a4964c9975252
2. The utility runs, replaces placeholders in all go files with new error codes starting from `next_err_code` and updates `component_info.json`: https://github.com/DelusionalOptimist/meshery/commit/ab8886758c438a6758b2c404bd000c7f0f713dce

Basically the utility is doing the hard work. The workflow just takes care of running the utility and committing the appropriate changes.

*NOTE: These tests were run using the master branch of MeshKit which contains a fix for the `component_info.json` bug.*

~If all these steps feel little over-engineered feel free to just use `.gitignore` :joy:.~

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
What an excellent write-up.

**Comment by DelusionalOptimist** [comment]
> What an excellent write-up.

Learning from @mgfeller :raised_hands:

**Comment by leecalcote** [comment]
@dhruv0000 @sayantan1413 @Shreyas220 @revolyssup, please be aware of this update.
